public/app/features/dashboard/components/ShareModal/SharePublicDashboard/ConfigPublicDashboard/SettingsBarHeader.tsx

```diff
@@ -0,0 +1,93 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { IconButton, ReactUtils, useStyles2 } from '@grafana/ui';
+
+export interface Props {
+  onRowToggle: () => void;
+  isContentVisible?: boolean;
+  title?: string;
+  headerElement?: React.ReactNode | ((props: { className?: string }) => React.ReactNode);
+}
+
+export function SettingsBarHeader({ headerElement, isContentVisible = false, onRowToggle, title }: Props) {
+  const styles = useStyles2(getStyles);
+
+  const headerElementRendered =
+    headerElement && ReactUtils.renderOrCallToRender(headerElement, { className: styles.summaryWrapper });
+
+  return (
+    <div className={styles.wrapper}>
+      <div className={styles.header}>
+        <IconButton
+          name={isContentVisible ? 'angle-down' : 'angle-right'}
+          tooltip={isContentVisible ? 'Collapse settings' : 'Expand settings'}
+          className={styles.collapseIcon}
+          onClick={onRowToggle}
+          aria-expanded={isContentVisible}
+        />
+        {title && (
+          // disabling the a11y rules here as the IconButton above handles keyboard interactions
+          // this is just to provide a better experience for mouse users
+          // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+          <div className={styles.titleWrapper} onClick={onRowToggle}>
+            <div className={styles.title}>{title}</div>
```

Comment:
shouldn't it be a <p> or <span> tag?

Reply:
It wouldn't make a difference in a flex layout, as it is now

Reply:
yes, but in order to be more precise and use the correct HTML tag, I think it should be one of those

Reply:
unfortunately, a p adds margins to itself, so I'll use a span